shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/documentservice/ExecutionDocumentService.java

```diff
@@ -127,17 +123,139 @@ private Document getUpdateOperationOnFullExecutionData(String executionData) thr
+                log.debug("updateDocument() :: database: " + "shepherd_graph_endpoint" + " and collection: " + "graph_endpoint"
+                        + " is document Updated :" + updateResult.wasAcknowledged());
+                boolean ack = updateResult.wasAcknowledged();
+                return ack;
+            }
+        } catch (MongoWriteException e) {
+            log.error(e.getMessage(), e);
+            throw e;
+        } catch (Exception ex) {
+            throw ex;
+        }
+        return false;
+    }
+
+    public ObjectId registerEndPointDetails(EndpointDetails endpointDetails) throws IOException {
+        if(this.mongoClient == null){
+            this.mongoClient = getMongoClient();
+        }
+        try {
+            MongoCollection<Document> collection = ExecutionDocumentServiceHelper.getMongoCollection(this.mongoClient, "shepherd_graph_endpoint", "graph_endpoint");
+
+            if (collection != null) {
+                final Document dbObjectInput = new Document();
+
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.CLIENT_ID,endpointDetails.getClientId());
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.ENDPOINT_NAME, endpointDetails.getEndpointName());
+                dbObjectInput.append(ExecutionDocumentConstants.Fields.ENDPOINT_DETAILS, JSONLoader.stringify(endpointDetails));
+                collection.insertOne(dbObjectInput);
+                log.debug(String.format("EndPoint details are inserted : %s inserted successfully in \n collection : %s and db : %s", endpointDetails, "graph_endpoint", "shepherd_graph_endpoint"));
+                return (ObjectId)dbObjectInput.get( "_id" );
+            }
+        } catch (MongoWriteException e) {
+            log.error(e.getMessage(), e);
+            throw e;
+        } catch (Exception ex) {
+            throw ex;
+        }
+        return null;
+    }
+
+
+
+    public EndpointDetails fetchEndPointDetails(Integer clientId, String endPointName) throws IOException {
+        if(this.mongoClient == null){
+            this.mongoClient = getMongoClient();
+        }
+        try {
+            MongoCollection<Document> collection = ExecutionDocumentServiceHelper.getMongoCollection(this.mongoClient, "shepherd_graph_endpoint", "graph_endpoint");
+            if (collection != null) {
+                Document result = collection.find(getSearchFilterForGraph_EndPointDetails(clientId, endPointName)).first();
+                if(result!= null){
+                    String endPointDetailsString = (String)result.get("endPointDetails");
+                    return JSONLoader.loadFromStringifiedObject(endPointDetailsString,EndpointDetails.class);
+                }
+            }
+        } catch (MongoWriteException e) {
+            log.error(e.getMessage(), e);
```

Comment:
improve logging. please use logging written as reference. Try to use single standard throughtout the service.